contxt/services/ems.py

```diff
@@ -1,102 +1,100 @@
-
-        response = self.execute(GET(uri=f"facilities/{facility_id}"), execute=True)
-
-        # manual filter for main services
-        rows = []
-        if type:
-            for row in response["main_services"]:
-                if row["type"] == type:
-                    rows.append(row)
-        else:
-            rows = response["main_services"]
-
-        return APIObjectCollection([FacilityMainService(row) for row in rows])
+    _envs = (
+        ApiEnvironment(
+            name="production",
+            base_url="https://ems.api.ndustrial.io/v1",
+            client_id="e2IT0Zm9RgGlDBkLa2ruEcN9Iop6dJAS",
+        ),
+        ApiEnvironment(
+            name="staging",
+            base_url="https://ems-staging.api.ndustrial.io/v1",
+            client_id="vMV67yaRFgjBB1JFbT3vXBOlohFdG1I4",
+        ),
+    )
+
+    def __init__(self, auth: Auth, env: str = "production"):
+        super().__init__(env, auth)
+
+    def get_facility(self, id: int) -> Facility:
+        return Facility.from_api(self.get(f"facilities/{id}"))
+
+    def get_main_services(
+        self, facility_id: int, resource_type: Optional[ResourceType] = None
+    ) -> List[MainService]:
+        facility = self.get_facility(facility_id)
+        main_services = facility.main_services
+
+        # Manually filter on resource type
+        if resource_type:
+            main_services = [
+                s
+                for s in main_services
+                if s.resource_type.lower() == resource_type.value.lower()
+            ]
+        return main_services
 
     def get_monthly_utility_spend(
         self,
         facility_id: int,
-        type: str,
-        date_start: datetime,
-        date_end: datetime,
+        resource_type: ResourceType = ResourceType.ELECTRIC,
+        start_date: date = date.today() - timedelta(days=3600),
+        end_date: date = date.today(),
         pro_forma: bool = False,
         exclude_account_charges: bool = False,
-    ):
-        params = {
-            "type": type,
-            "date_start": date_start.strftime("%Y-%m"),
-            "date_end": date_end.strftime("%Y-%m"),
-            "pro_forma": "true" if pro_forma else "false",
-            "exclude_account_charges": "true" if exclude_account_charges else "false",
-        }
-
-        response = self.execute(
-            GET(uri=f"facilities/{facility_id}/utility/spend/monthly").params(params),
-            execute=True,
+    ) -> UtilitySpend:
+        """Get monthly utility spend for facility `facility_id` and resource
+        `resource_type` (for now, this must be "electric" but will be expanded).
+        Note `start_date` defaults to 10 years ago (the API's maximum limit) and
+        `end_date` defaults to today."""
+        resp = self.get(
+            f"facilities/{facility_id}/utility/spend/monthly",
+            params={
+                "type": resource_type.value,
```

Comment:
I wonder if the endpoint before this change assumed the type was of "ELECTRIC"?  Or maybe the param was so badly named it was hard to tell

Reply:
before, the `type` parameter accepted any string. however, the API currently only supports `electric`